src/prefect/utilities/agent.py

```diff
@@ -1,9 +1,21 @@
 from distutils.version import LooseVersion
+from typing import Optional
 
 import prefect
 from prefect.utilities.graphql import GraphQLResult
 
 
+def get_flow_image_if_docker_storage(flow_run: GraphQLResult) -> Optional[str]:
```

Comment:
Why introduce a new function here instead of altering `get_flow_image`?

Reply:
The logic is much simpler than that of `get_flow_image`, and only pertains to flows using docker storage. I also want to know (without an exception) if no image is configured, `get_flow_image` raises if no image. I could merge them, but I'd want to change the interface of `get_flow_image` and didn't want to track down all uses of it and update them if not necessary.